### PR TITLE
Loading screen and animated Content Studio Logo (with CSS)

### PR DESCRIPTION
**Pull request by kristofferj**

### CODE REVIEW

File: packages/@sanity/default-layout/src/components/DefaultLayout.js [hunk 1]
```diff
@@ -61,6 +72,16 @@ export default withRouterHOC(class DefaultLayout extends React.Component {
     })
   }
 
+  setLoadingScreenElement = element => {
+    if (element && this.state.showLoadingScreen) {
+      element.addEventListener('animationend', event => {
```
Comment (rexxars) [inline packages/@sanity/default-layout/src/components/DefaultLayout.js]: This method should be assigned on the class instance, and should have a `componentWillUnmount` method that removes the event listener.

File: packages/@sanity/default-login/src/LoginWrapper.js [hunk 2]
```diff
@@ -30,13 +30,15 @@ export default class LoginWrapper extends React.PureComponent {
...
-    sanityLogo: PropTypes.node
+    sanityLogo: PropTypes.node,
+    loadingScreen: PropTypes.node,
   }
 
   static defaultProps = {
     title: 'Choose login provider',
     description: null,
     sanityLogo: <SanityStudioLogo />,
+    loadingScreen: <Spinner center fullscreen />
```
Comment (rexxars) [inline packages/@sanity/default-login/src/LoginWrapper.js]: I think both of these props (`sanityLogo` and `loadingScreen`) should be components, node actual nodes?

File: packages/@sanity/default-layout/src/components/DefaultLayout.js (hunk 1 above)
Comment (kristofferj) [reply packages/@sanity/default-layout/src/components/DefaultLayout.js]: 👍

File: packages/@sanity/default-login/src/LoginWrapper.js (hunk 2 above)
Comment (kristofferj) [reply packages/@sanity/default-login/src/LoginWrapper.js]: Hm maybe `PropTypes.node` is wrong. `PropTypes.element` does not give any error.  Or should it be on only functions?

Comment (skogsmaskin) [reply packages/@sanity/default-login/src/LoginWrapper.js]: Please note that this is a breaking change, and things around needs to be updated if the sanityLogo proptype is changed.

Comment (skogsmaskin) [reply packages/@sanity/default-login/src/LoginWrapper.js]: Example where we use this:
https://github.com/sanity-io/manage-sanity-io/blob/26f4ff5c6a09f6604a8214e71448fb902c921b7d/src/routes/invite/Invite.js#L191

Comment (rexxars) [reply packages/@sanity/default-login/src/LoginWrapper.js]: I don't think it's a great idea to pass "pre-rendered" elements around, but if we want to maintain backwards compatibility we could do something along the lines of:

```js
const LoginWrapper = () => (
  <div>
    {/* ... */}
    {typeof LoadingScreen === 'function' ? <LoadingScreen /> : LoadingScreen}
  </div>
)
```

Then set the prop type to `PropTypes.oneOfType([PropTypes.func, PropTypes.node])`. Thoughts?

**Pull request merged**